src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -149,10 +155,33 @@ final class Region(
 
   private def allocate(n: Long): Long = {
     assert(n >= 0)
-    val off = end
-    ensure(n)
-    end += n
-    off
+    if (n > blockSize) {
+      // FIXME: is this guaranteed to be aligned to anything?
+      log.info(s"Allocating a large region $n")
+      val mem = Memory.malloc(n)
+      bigBlocks += mem
+      mem
+    } else {
+      val required = end + n
+      if (required > capacity) {
+        // FIXME: is this guaranteed to be aligned to anything?
+        activeBlock += 1
+        val mem = if (activeBlock == blocks.length) {
+          log.info(s"Allocating a new region, $end $n $capacity")
+          val temp = Memory.malloc(blockSize)
+          blocks += temp
+          temp
+        } else {
+          blocks(activeBlock)
+        }
+        end = n
+        mem
+      } else {
+        val mem = blocks(activeBlock) + end
+        end += n
+        mem
+      }
+    }
   }
 
   def allocate(alignment: Long, n: Long): Long = {
```

Comment:
This is a bit flaky, because the allocate() isn't necessarily going to give you something in the same
buffer that you just align'ed.  I guess it kinda sorta works as long as the start-of-some-new-buffer
is always sufficiently well aligned, but that's relying on some not-very-well-specified behavior.